topic-operator/src/main/java/io/strimzi/operator/topic/v2/BatchingTopicController.java

```diff
@@ -0,0 +1,989 @@
+                return Either.ofRight(e);
+            }
+
+        }
+        return Either.ofLeft(true);
+    }
+
+    /* test */ static boolean matchesSelector(Map<String, String> selector, Map<String, String> resourceLabels) {
+        if (!selector.isEmpty()) {
+            for (var selectorEntry : selector.entrySet()) {
+                String resourceValue = resourceLabels.get(selectorEntry.getKey());
+                if (resourceValue == null
+                        || !resourceValue.equals(selectorEntry.getValue())) {
+                    return false;
+                }
+            }
+        }
+        return resourceLabels.keySet().containsAll(selector.keySet());
+    }
+
+    private static Either<Boolean, TopicOperatorException> validateUnchangedTopicName(ReconcilableTopic reconcilableTopic) {
+        if (reconcilableTopic.kt().getStatus() != null
+                && reconcilableTopic.kt().getStatus().getTopicName() != null
+                && !topicName(reconcilableTopic.kt()).equals(reconcilableTopic.kt().getStatus().getTopicName())) {
+            return Either.ofRight(new TopicOperatorException.NotSupported("Changing spec.topicName is not supported"
+            ));
+        }
+        return Either.ofLeft(true);
+    }
+
+    private PartitionedByError<ReconcilableTopic, Void> createTopics(List<ReconcilableTopic> kts) {
+        var newTopics = kts.stream().map(reconcilableTopic -> {
+            // Admin create
+            return buildNewTopic(reconcilableTopic.kt(), reconcilableTopic.topicName());
+        }).collect(Collectors.toSet());
+
+        LOGGER.debugOp("Admin.createTopics({})", newTopics);
+        long t0 = System.nanoTime();
+        CreateTopicsResult ctr = admin.createTopics(newTopics);
+        if (LOGGER.isTraceEnabled()) {
+            ctr.all().whenComplete((i, e) -> {
+                if (e != null) {
+                    LOGGER.traceOp("Admin.createTopics({}) took {}ns to fail with {}", newTopics, System.nanoTime() - t0, String.valueOf(e));
+                } else {
+                    LOGGER.traceOp("Admin.createTopics({}) took {}ns", newTopics, System.nanoTime() - t0);
+                }
+
+            });
+        }
+        Map<String, KafkaFuture<Void>> values = ctr.values();
+        return partitionedByError(kts.stream().map(reconcilableTopic -> {
+            try {
+                values.get(reconcilableTopic.topicName()).get();
+                return pair(reconcilableTopic, Either.ofLeft((null)));
+            } catch (ExecutionException e) {
+                return pair(reconcilableTopic, Either.ofRight(handleAdminException(e)));
+            } catch (InterruptedException e) {
+                throw new UncheckedInterruptedException(e);
+            }
+        }));
+    }
+
+    private static TopicOperatorException handleAdminException(ExecutionException e) {
+        // Note this method is declared to return a RuntimeException purely so that the caller
```

Comment:
Comment states RuntimeException, but the TopicOperatorException extends Exception?